### ISSUE TITLE
Adjust `internal/scripts/generate-coverage.sh` to use source-coverage

### ISSUE THREAD

**Issue by elfenpiff** [eclipse-iceoryx/iceoryx2#372]
## Brief feature description

The current code coverage script is not yet perfectly configured to generate a branch coverage report.
Follow the documentation from:
 * https://doc.rust-lang.org/rustc/instrument-coverage.html
 * https://rustc-dev-guide.rust-lang.org/llvm-coverage-instrumentation.html

To adjust the script so that it generates a coverage report
 * for the whole iceoryx2 workspace
 * for every component individually, via CLI arguments
 * for the C and C++ language bindings.

**Comment by Afshankk** [comment]
Hi, I'd like to work on this issue. Is it still relevant?
If so, I’ll prepare a PR. Thanks!

**Comment by elfenpiff** [comment]
@Afshankk, yes, it is still relevant, and it would be great if you want to work on this issue. Thanks for the offer! If you have questions, please do not hesitate to ask.
